# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.h
# Commit: c2233b3830ef
# Full Hash: c2233b3830efd3e475a76443d22c07e73cda476a
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-06-11 09:51:55
# Regressor Bug: 1758689
# File Overlap Count: 1
# Description:
#   Bug 1758689: Create and maintain a viewport cache for hittesting r=Jamie,emilio
#   
#   This hittesting approach is borrowed from our Android implementation.
#   We maintain a cache of ID's for accessibles which are visible in the viewport.
#   This cache is created with a call to `nsLayoutUtils::GetFramesForArea`, which
# ==============================================================================

diff -r 7a17841a34cc -r c2233b3830ef accessible/generic/DocAccessible.h
--- a/accessible/generic/DocAccessible.h	Fri Jun 10 23:49:24 2022 +0300
+++ b/accessible/generic/DocAccessible.h	Fri Jun 10 21:10:18 2022 +0000
@@ -153,6 +153,9 @@
 
   bool IsHidden() const;
 
+  bool IsViewportCacheDirty() { return mViewportCacheDirty; }
+  void SetViewportCacheDirty(bool aDirty) { mViewportCacheDirty = aDirty; }
+
   /**
    * Document load states.
    */
@@ -662,7 +665,15 @@
   /**
    * Bit mask of other states and props.
    */
-  uint32_t mDocFlags : 28;
+  uint32_t mDocFlags : 27;
+
+  /**
+   * Tracks whether we have seen changes to this document's content that
+   * indicate we should re-send the viewport cache we use for hittesting.
+   * This value is set in `BundleFieldsForCache` and processed in
+   * `ProcessQueuedCacheUpdates`.
+   */
+  bool mViewportCacheDirty : 1;
 
   /**
    * Type of document load event fired after the document is loaded completely.